client/rest-high-level/src/test/java/org/elasticsearch/client/SnapshotIT.java

```diff
@@ -108,4 +112,18 @@ public void testVerifyRepository() throws IOException {
             highLevelClient().snapshot()::verifyRepositoryAsync);
         assertThat(response.getNodes().size(), equalTo(1));
     }
+
+    private CreateSnapshotResponse createTestSnapshot(String repository, String snapshot) throws IOException {
+        // assumes the repository already exists
+        CreateSnapshotRequest request = new CreateSnapshotRequest(repository, snapshot);
```

Comment:
I think we should add more of the options here, randomly with some `if randomBoolean()`

Reply:
Added several random booleans for options where it made sense.